lib/api/actions/functions.php

```diff
@@ -410,119 +411,184 @@ function beans_reset_action( $id ) {
+	$registered_action = _beans_get_action( $id, $status );
+
+	// If the action is registered and we are not overwriting, return it.
+	if ( true !== $overwrite && ! empty( $registered_action ) ) {
+		return $registered_action;
 	}
 
-	$_beans_registered_actions[ $status ][ $id ] = json_encode( $action );
+	// Let's set (or overwrite) the action.
+	global $_beans_registered_actions;
+	$_beans_registered_actions[ $status ][ $id ] = wp_json_encode( $action );
 
 	return $action;
-
 }
 
 /**
- * Unset action.
+ * Unset the action's configuration for the given ID and status. Returns `false` if there are is no action
+ * registered with Beans actions for the given ID and status. Else, returns true when complete.
  *
+ * @since  1.5.0
  * @ignore
+ * @access private
+ *
+ * @param string $id     A unique string used as a reference.
+ * @param string $status Status for which to get the action.
+ *
+ * @return bool
  */
 function _beans_unset_action( $id, $status ) {
-
-	global $_beans_registered_actions;
-
 	$id = _beans_unique_action_id( $id );
 
-	// Stop here if the action doesn't exist.
-	if ( ! _beans_get_action( $id, $status ) ) {
+	// Bail out if the ID is not registered for the given status.
+	if ( false === _beans_get_action( $id, $status ) ) {
 		return false;
 	}
 
+	global $_beans_registered_actions;
 	unset( $_beans_registered_actions[ $status ][ $id ] );
 
 	return true;
-
 }
 
 /**
- * Merge action.
+ * Merge the action's configuration and then store it for the given ID and status.
  *
+ * If the action's configuration has not already be registered with Beans, just store it.
+ *
+ * @since  1.5.0
  * @ignore
+ * @access private
+ *
+ * @param string $id     A unique string used as a reference.
+ * @param array  $action The new action's configuration to merge and then store.
+ * @param string $status Status for which to merge/store this action.
+ *
+ * @return array
  */
-function _beans_merge_action( $id, $action, $status ) {
-
-	global $_beans_registered_actions;
-
-	$id = _beans_unique_action_id( $id );
+function _beans_merge_action( $id, array $action, $status ) {
+	$id                = _beans_unique_action_id( $id );
+	$registered_action = _beans_get_action( $id, $status );
 
-	if ( $_action = _beans_get_action( $id, $status ) ) {
-		$action = array_merge( $_action, $action );
+	// If the action's configuration is already registered with Beans, merge the new configuration with it.
+	if ( false !== $registered_action ) {
+		$action = array_merge( $registered_action, $action );
 	}
 
+	// Now store/register it.
 	return _beans_set_action( $id, $action, $status, true );
-
 }
 
 /**
- * Check all action status and return the current action.
+ * Get the current action, meaning get from the "added" and/or "modified" statuses.
  *
+ * @since  1.5.0
  * @ignore
+ * @access private
+ *
+ * @param string $id A unique string used as a reference.
+ *
+ * @return array|bool
  */
 function _beans_get_current_action( $id ) {
```

Comment:
Ignore this function.  It's part of PR #92.  It's shown here as I built this PR off of PR #92's branch.